packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerOptions.kt

```diff
@@ -1,58 +1,119 @@
-          return@let it
-        }
-
-        promise.reject(ImagePickerConstants.ERR_INVALID_OPTION, "'Aspect option must be of form [Number, Number]")
-        return null
-      }
+import android.net.Uri
+import android.provider.MediaStore
+import androidx.annotation.FloatRange
+import androidx.annotation.IntRange
+import expo.modules.imagepicker.crop.CameraContract
+import expo.modules.imagepicker.crop.ImageLibraryContract
+import expo.modules.kotlin.assertions.assertValueGreaterOrEqual
+import expo.modules.kotlin.assertions.assertValueInRange
+import expo.modules.kotlin.records.Field
+import expo.modules.kotlin.records.Record
+
+internal class ImagePickerOptions: Record {
+  @Field
+  var allowsEditing: Boolean = false
+
+  @Field
+  var allowsMultipleSelection: Boolean = false
+
+  @Field
+  @FloatRange(from = 0.0, to = 1.0)
+  var quality: Double = 0.2
+    set(value) {
+      // TODO(@bbarthec): possibly we don't need that
+      assertValueInRange(value, lowerBound = 0.0, upperBound = 1.0)
+      field = value
+    }
+
+  @Field
+  var base64: Boolean = false
 
-      val isBase64 = options[ImagePickerConstants.OPTION_BASE64] as? Boolean ?: false
-      val mediaTypes = MediaTypes.fromString(
-        options[ImagePickerConstants.OPTION_MEDIA_TYPES] as? String ?: "Images"
-      ).ifNull {
-        promise.reject(ImagePickerConstants.ERR_INVALID_OPTION, "Unknown media types: ${options[ImagePickerConstants.OPTION_MEDIA_TYPES]}")
-        return null
+  @Field
+  var exif: Boolean = false
+
+  @Field
+  var mediaTypes: MediaTypes = MediaTypes.IMAGES
+
+  @Field
+  @IntRange(from = 0)
+  var videoMaxDuration: Int = 0
+    set(value) {
+      assertValueGreaterOrEqual(value, 0)
+      field = value
+    }
+
+  @Field
+  var aspect: Pair<Int, Int>? = null
+    set(value) {
+      // TODO(@bbarthec): check if that works
+      if (value != null) {
+        val (first, second) = value
+        assertValueGreaterOrEqual(first, 0)
+        assertValueGreaterOrEqual(second, 0)
       }
-      val isExif = options[ImagePickerConstants.OPTION_EXIF] as? Boolean ?: false
+      field = value
+    }
+
+  /**
+   * @param uri Target [Uri] that the captured media will be saved into.
+   */
+  internal fun toCameraContract(uri: Uri) = CameraContract(
+    uri,
+    mediaTypes.toMimeType(),
+  )
 
-      val videoMaxDuration = options[ImagePickerConstants.OPTION_VIDEO_MAX_DURATION]?.let {
-        if (it is Number && it.toInt() >= 0) {
-          return@let it.toInt()
-        }
+  fun toImageLibraryContract() = ImageLibraryContract(
+    allowsMultipleSelection,
+    mediaTypes.toMimeType(),
+    mediaTypes.toMultipleMimeTypes()
+  )
+}
+
+enum class MediaTypes(val value: String) {
+  IMAGES("Images"),
+  VIDEOS("Videos"),
+  ALL("All");
+
+  internal fun toMimeType(): String {
+    return when (this) {
+      IMAGES -> ImageAllMimeType
+      VIDEOS -> VideoAllMimeType
+      ALL -> AllMimeType
+    }
+  }
```

Comment:
I prefer this syntax, but up to you
```suggestion
  internal fun toMimeType(): String = when (this) {
    IMAGES -> ImageAllMimeType
    VIDEOS -> VideoAllMimeType
    ALL -> AllMimeType
  }
```